zrml/prediction-markets/src/tests.rs

```diff
@@ -427,6 +428,64 @@ fn admin_destroy_market_correctly_slashes_permissionless_market_disputed() {
     });
 }
 
+#[test]
+fn admin_destroy_market_correctly_unreserves_dispute_bonds() {
+    ExtBuilder::default().build().execute_with(|| {
+        let end = 2;
+        simple_create_categorical_market(MarketCreation::Permissionless, 0..end, ScoringRule::CPMM);
+        let market = MarketCommons::market(&0).unwrap();
+        let grace_period = end + market.deadlines.grace_period;
+        assert_ne!(grace_period, 0);
+        run_to_block(grace_period + 1);
+        assert_ok!(PredictionMarkets::report(
+            Origin::signed(BOB),
+            0,
+            OutcomeReport::Categorical(1)
+        ));
+        run_to_block(grace_period + 2);
+        assert_ok!(PredictionMarkets::dispute(
+            Origin::signed(CHARLIE),
+            0,
+            OutcomeReport::Categorical(0)
+        ));
+        assert_ok!(PredictionMarkets::dispute(
+            Origin::signed(DAVE),
+            0,
+            OutcomeReport::Categorical(1)
+        ));
+        let set_up_account = |account| {
+            assert_ok!(AssetManager::deposit(Asset::Ztg, account, SENTINEL_AMOUNT));
+            assert_ok!(Balances::reserve_named(
+                &PredictionMarkets::reserve_id(),
+                account,
+                SENTINEL_AMOUNT,
+            ));
+        };
+        set_up_account(&CHARLIE);
+        set_up_account(&DAVE);
+
+        let balance_free_before_charlie = Balances::free_balance(&CHARLIE);
+        let balance_free_before_dave = Balances::free_balance(&DAVE);
+        assert_ok!(PredictionMarkets::admin_destroy_market(Origin::signed(SUDO), 0));
```

Comment:
A check that `Disputes` at market id 0 is actually empty could be useful here. But not so important after ZIP-0.

Reply:
[f6bbfb3](https://github.com/zeitgeistpm/zeitgeist/pull/877/commits/f6bbfb3a3371d3b0774f01b5d49763d842433c34)